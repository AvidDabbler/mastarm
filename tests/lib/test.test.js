/* globals describe, it, expect */

const testUtils = require('../../lib/test')

describe('test.js', () => {
  it('generateTestConfig should generate proper config', () => {
    const cfg = testUtils.generateTestConfig({
      coveragePaths: 'bin src another-folder',
      updateSnapshots: true,
      cache: false
    })
    expect(cfg).toBeTruthy()
    expect(cfg.length).toEqual(4)
    expect(cfg.slice(0, 3)).toMatchSnapshot()
    const jestCfg = JSON.parse(cfg[3])
    expect(jestCfg.scriptPreprocessor).toContain('lib/jestPreprocessor.js')
    delete jestCfg.scriptPreprocessor
    expect(jestCfg).toMatchSnapshot()
  })

  it('setupTestEnvironment should set MESSAGES environment variable if needed', () => {
    testUtils.setupTestEnvironment({ messages: { hi: 'there' } })
    expect(process.env.MESSAGES).toMatchSnapshot()
  })
})
