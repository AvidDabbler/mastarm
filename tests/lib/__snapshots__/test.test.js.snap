exports[`test.js generateTestConfig should generate proper config 1`] = `
Array [
  "-u",
  "--no-cache",
  "--config"
]
`;

exports[`test.js generateTestConfig should generate proper config 2`] = `
Object {
  "cacheDirectory": "tmp",
  "collectCoverageFrom": Array [
    "lib/**/*.js",
    "bin",
    "src",
    "another-folder"
  ],
  "coverageDirectory": "coverage"
}
`;

exports[`test.js setupTestEnvironment should set MESSAGES environment variable if needed 1`] = `"{\"hi\":\"there\"}"`;
